iml-config/src/nginx.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlConfigError;
+use console::Term;
+use std::{env, str};
+use structopt::StructOpt;
+use tokio::fs;
+
+#[derive(Debug, StructOpt)]
+pub enum NginxCommand {
+    /// Generate Nginx conf
+    #[structopt(name = "generate-config")]
+    GenerateConfig {
+        /// Set the nginx config path
+        #[structopt(short = "p", long = "path")]
+        template_path: String,
+
+        #[structopt(short = "o", long = "output")]
+        output_path: Option<String>,
+    },
+}
+
+const TEMPLATE_VARS: &[&str] = &[
+    "REPO_PATH",
+    "HTTP_FRONTEND_PORT",
+    "HTTPS_FRONTEND_PORT",
+    "HTTP_AGENT_PROXY_PASS",
+    "HTTP_AGENT2_PROXY_PASS",
+    "HTTP_API_PROXY_PASS",
+    "IML_API_PROXY_PASS",
+    "WARP_DRIVE_PROXY_PASS",
+    "MAILBOX_PATH",
+    "MAILBOX_PROXY_PASS",
+    "SSL_PATH",
+    "DEVICE_AGGREGATOR_PORT",
+    "DEVICE_AGGREGATOR_PROXY_PASS",
+    "UPDATE_HANDLER_PROXY_PASS",
+    "GRAFANA_PORT",
+    "GRAFANA_PROXY_PASS",
+    "INFLUXDB_PROXY_PASS",
+    "TIMER_PROXY_PASS",
+    "INCLUDES",
+];
```

Comment:
We need to find another way to do this. We'll need to keep at least three lists in sync (this one, settings.py, print-settings.py

Reply:
removed :)